utilities/tools/Test-ModuleLocally.ps1

```diff
@@ -118,106 +126,121 @@ function Test-ModuleLocally {
+            }
+
+            # Replace parameter file tokens
+            # -----------------------------
+
+            # Default Tokens
+            $ConvertTokensInputs = @{
+                Tokens = @{
+                    subscriptionId    = $ValidateOrDeployParameters.SubscriptionId
+                    managementGroupId = $ValidateOrDeployParameters.ManagementGroupId
                 }
-                #Add Other Parameter File Tokens (For Testing)
-                if ($AdditionalTokens) {
-                    $ConvertTokensInputs += @{ OtherCustomParameterFileTokens = $AdditionalTokens
-                    }
+            }
+
+            #Add Other Parameter File Tokens (For Testing)
+            if ($AdditionalTokens) {
+                $ConvertTokensInputs.Tokens += $AdditionalTokens
+            }
+
+            # Tokens in settings.json
+            $settingsFilePath = Join-Path (Get-Item $PSScriptRoot).Parent.Parent 'settings.json'
+            if (Test-Path $settingsFilePath) {
+                $Settings = Get-Content -Path $settingsFilePath -Raw | ConvertFrom-Json -AsHashtable
+                $ConvertTokensInputs += @{
+                    TokenPrefix = $Settings.parameterFileTokens.tokenPrefix
+                    TokenSuffix = $Settings.parameterFileTokens.tokenSuffix
+                }
+
+                if ($Settings.parameterFileTokens.localTokens) {
+                    $ConvertTokensInputs.Tokens += $Settings.parameterFileTokens.localTokens
                 }
             }
+
             # Invoke Token Replacement Functionality and Convert Tokens in Parameter Files
-            $ModuleParameterFiles | ForEach-Object { $null = Convert-TokensInParameterFile @ConvertTokensInputs -ParameterFilePath $PSItem.FullName }
-            # Build Modules Validation and Deployment Inputs
+            $ModuleParameterFiles | ForEach-Object { $null = Convert-TokensInParameterFile @ConvertTokensInputs -ParameterFilePath $_ }
+
+            # Deployment & Validation Testing
+            # -------------------------------
             $functionInput = @{
                 templateFilePath  = $templateFilePath
-                parameterFilePath = (Join-Path (Split-Path $templateFilePath -Parent) '.parameters/parameters.json')
-                location          = "$($ValidateOrDeployParameters.Location)"
-                resourceGroupName = "$($ValidateOrDeployParameters.ResourceGroupName)"
-                subscriptionId    = "$($ValidateOrDeployParameters.SubscriptionId)"
-                managementGroupId = "$($ValidateOrDeployParameters.ManagementGroupId)"
+                location          = $ValidateOrDeployParameters.Location
+                resourceGroupName = $ValidateOrDeployParameters.ResourceGroupName
+                subscriptionId    = $ValidateOrDeployParameters.SubscriptionId
+                managementGroupId = $ValidateOrDeployParameters.ManagementGroupId
+                Verbose           = $true
```

Comment:
just for my own knowledge, do we need to force verbose here? if we run the overarching script as -verbose?

Reply:
This verbose is supposed to ensure the `Test-Az*Deployment` and `New-Az*Deployment` always return a result - not matter if the script itself was run with `verbose`